src/org/omegat/filters2/html2/FilterVisitor.java

```diff
@@ -82,11 +84,16 @@ public FilterVisitor(HTMLFilter2 htmlfilter, BufferedWriter bufwriter, HTMLOptio
     // Variable declaration
     // ///////////////////////////////////////////////////////////////////////
 
-    /** Should the parser call us for this tag's ending tag and its inner tags. */
-    protected boolean recurse = true;
+    /** Should the parser call us for this tag's ending tag. */
```

Comment:
This change fixes the bug. Most of the other changes are renaming variables to make the code clearer.